src/audio/dai.c

```diff
@@ -114,51 +113,6 @@ static void dai_dma_cb(void *data, uint32_t type, struct dma_sg_elem *next)
 		return;
 	}
 
-	if (dev->params.direction == SOF_IPC_STREAM_PLAYBACK) {
```

Comment:
The pointers must be updated where they are copied, this delays the update for DAI meaning the DAI R/W can become out of sync with the HW before the pipeline is scheduled. This fix should probably go in buffer APIs to better cope with DMA position updates that are async with pipeline scheduling.

Please log an issues for this and include logs and any other relevant details.

Reply:
I did the test with this patch on the UP^2 platform. it can fix the issue: 
pause capture when playback is runing will fail #22.
But this patch can ONLY fix this on UP^2 platform. I tried it on GP. It still has this bug.
I think there still other factor will cause this DSP panic.
Maybe UP^2 and GP has something different. 
During this test, I have to check whether there is an kernel XRUN happen in GP or not when I am back to office. 
During this test, I did not find kernel XRUN On UP^2.

About the pointer update sync, from other side, the HW and buffer pointer are updated at the same time. because during the pipeline is running, the dma is still running.